### ISSUE TITLE
Problem with raw IP endpoints

### ISSUE THREAD

**Issue by ggprod** [StackExchange/StackExchange.Redis#1399]
I recently debugged an issue connecting a .NET service using Microsoft.Extensions.Caching.StackExchangeRedis v2.2.5 (which uses StackExchange.Redis  2.0.593).  The client could not connect to a Google Cloud MemoryStore managed Redis instance (these provide only private IP addresses for connectivity).  It turned out the issue could be resolved by creating a name alias for the IP address and using the name instead.  This seems strange and I'd expect raw IP address endpoints to work normally.

**Comment by mgravell** [comment]
ipv4? Or ipv6?

**Comment by mgravell** [comment]
To expand on that: I haven't seen any problems with ipv4, so I'm speculating that you're using ipv6 and that there is some ipv6-specific handling bug.

**Comment by ggprod** [comment]
No, it was IPv4, but perhaps the bug is in the Microsoft.Extensions.Caching.StackExchangeRedis if you say you don't see the issue using just the StackExchange.Redis package

**Comment by mgravell** [comment]
I haven't *seen* an issue, indeed. That doesn't mean there isn't one. But: switching from DNS to IP has been a successful hack in some cases when DNS resolution has been unreliable.

As with most things, the closer we can get to a reliable repro; the better it is for us investigating

**Comment by ggprod** [comment]
Sure, ok, in this case it was a .NET service running in a Docker container deployed (as a Deployment of Pods) to Google Kubernetes Engine.  The Docker base image is mcr.microsoft.com/dotnet/core/runtime:3.1 and the application interacts with Redis through the Microsoft.Extensions.Caching.StackExchangeRedis v2.2.5 (which uses StackExchange.Redis 2.0.593).  The service was initially configured to pass the redis endpoint in a form like 10.101.192.188:6379 and was not able to connect to the Redis instance.  I verified that I was able to connect to the MemoryStore instance on the same IP/port from on of the Deployment pods using the redis-cli.  We then tried giving a name to the IP address via hostAlias in the PodSpec (static.redis.host) and using that name as the endpoint passed and then connectivity to the instance became possible.
